recipes-kernel/linux/linux-imx-headers_5.4.bb

```diff
@@ -7,15 +7,18 @@ New headers are installed in ${includedir}/imx."
 LICENSE = "GPLv2"
 LIC_FILES_CHKSUM = "file://COPYING;md5=bbea815ee2795b2f4230826c0c6b8814"
 
-SRCBRANCH = "lf-5.4.y"
-LOCALVERSION = "-1.0.0"
-SRC_URI = "git://source.codeaurora.org/external/imx/linux-imx.git;protocol=https;branch=${SRCBRANCH}"
-SRCREV = "f8118585ee3c7025265b28985fdfe0af96a84466"
+# Inherit class that provides the kernel source.
+# This allows to switch kernels from NXP to combined NXP+fslc and not keep
+# fixed NXP-only repository to fetch headers, thus making it consistent
+# accross kernel trees chosen for the build.
+inherit kernelsrc
```

Comment:
I understand the motivation about this however, we cannot change this across machines.

So this is the reason we had this "locked" under a revision. This provides a ABI and changing it means all packages using this (and its dependencies) become `MACHINE_ARCH`.

Reply:
Would there be any reason to install linux-imx-headers but let's say - linux-qoriq?
Is a sane "way out" maybe installing linux-xxx-headers with linux-xxx and just add such a package?

Reply:
@rehsack we do not wish to replace common headers and we wish to reduce the headers installed here.

Doing so, we reduce the specific ABI we maintain so we don't make all packages machine specific. We also try to restrict the packages adding this so we have less specific packages.

Do you see a different approach?

Reply:
> Doing so, we reduce the specific ABI we maintain so we don't make all packages machine specific. We also try to restrict the packages adding this so we have less specific packages.

Agree, this is a very valid point.

I guess handling of this via `use-imx-headers.bbclass` ensures that recipes installing imx headers are contracting themselves to use NXP BSP only.

The next step would be to actually prohibit all recipes to add `linux-imx-headers` directly into their `DEPENDS` and rather inherit the class, which should IMHO complete the isolation.

> Do you see a different approach?

Actually, the cleanest way for me would be that NXP provides those headers in the archive (similar to how `linux-libc-headers` are provided), and make sure they do match to the kernel tree they release. But until this happens - fetching the whole NXP tree seems to be inevitable...

Reply:
> @rehsack we do not wish to replace common headers and we wish to reduce the headers installed here.

I'm probably to far away from this specific problem ...

I understand it that-wise: linux-imx-headers are required to build against linux-imx-kernel some userspace executables - likely on target (otherwise the recipe itself can use `kernelsrc.bbclass`).
Consequently - on such a target with imx kernels, the linux-imx-headers are part of the linux-headers (or better: should be). Am I wrong with this conclusion?

Reply:
> I understand it that-wise: linux-imx-headers are required to build against linux-imx-kernel some userspace executables - likely on target (otherwise the recipe itself can use `kernelsrc.bbclass`).
> Consequently - on such a target with imx kernels, the linux-imx-headers are part of the linux-headers (or better: should be). Am I wrong with this conclusion?

On an ideal world, the imx-headers would not exist. The current approach was the better balance I could come with as we reduce the number of rebuilds.

Any idea how to improve this even more?